owncloudApp/src/main/java/com/owncloud/android/ui/activity/FileDisplayActivity.kt

```diff
@@ -0,0 +1,1838 @@
+
+    private fun updateFragmentsVisibility(existsSecondFragment: Boolean) {
+        if (existsSecondFragment) {
+            if (leftFragmentContainer!!.visibility != View.GONE) {
+                leftFragmentContainer!!.visibility = View.GONE
+            }
+            if (rightFragmentContainer!!.visibility != View.VISIBLE) {
+                rightFragmentContainer!!.visibility = View.VISIBLE
+            }
+
+        } else {
+            if (leftFragmentContainer!!.visibility != View.VISIBLE) {
+                leftFragmentContainer!!.visibility = View.VISIBLE
+            }
+            if (rightFragmentContainer!!.visibility != View.GONE) {
+                rightFragmentContainer!!.visibility = View.GONE
+            }
+        }
+    }
+
+    private fun cleanSecondFragment() {
+        val second = secondFragment
+        if (second != null) {
+            val tr = supportFragmentManager.beginTransaction()
+            tr.remove(second)
+            tr.commit()
+        }
+        updateFragmentsVisibility(false)
+        updateActionBarTitleAndHomeButton(null)
+    }
+
+    protected fun refreshListOfFilesFragment(reloadData: Boolean) {
+        val fileListFragment = listOfFilesFragment
+        fileListFragment?.listDirectory(reloadData)
+    }
+
+    override fun onPrepareOptionsMenu(menu: Menu): Boolean {
+        val drawerOpen = isDrawerOpen
+        menu.findItem(R.id.action_sync_account).isVisible = !drawerOpen
+        menu.findItem(R.id.action_switch_view).isVisible = !drawerOpen
+
+        return super.onPrepareOptionsMenu(menu)
+    }
+
+    override fun onCreateOptionsMenu(menu: Menu): Boolean {
+        val inflater = menuInflater
+
+        // Allow or disallow touches with other visible windows
+        val actionBarView = findViewById<View>(R.id.action_bar)
+        if (actionBarView != null) {
+            actionBarView.filterTouchesWhenObscured =
+                PreferenceUtils.shouldDisallowTouchesWithOtherVisibleWindows(applicationContext)
+        }
+
+        inflater.inflate(R.menu.main_menu, menu)
+        inflater.inflate(R.menu.sort_menu, menu.findItem(R.id.action_sort).subMenu)
+        menu.findItem(R.id.action_create_dir).isVisible = false
+
+        descendingMenuItem = menu.findItem(R.id.action_sort_descending)
+        selectAllMenuItem = menu.findItem(R.id.action_select_all)
+        if (secondFragment == null) {
+            selectAllMenuItem!!.isVisible = true
+        }
+        mainMenu = menu
+
+        recoverSortMenuFormPreferences(menu)
+
+        return true
+    }
+
+    override fun onOptionsItemSelected(item: MenuItem): Boolean {
+
+        when (item.itemId) {
+            R.id.action_select_all -> {
+                listOfFilesFragment!!.selectAll()
+            }
+            R.id.action_sync_account -> {
+                startSynchronization()
+            }
+            android.R.id.home -> {
+                val second = secondFragment
+                val currentDir = currentDir
```

Comment:
@davigonz @abelgardep 
Such a construct should be avoided

Reply:
Ok, but I think it is too late for the code review. If you think it could be improved, create a new PR or open a new issue. Thanks!

Reply:
It's just a note to hopefully remember in the future. Nothing more